### PR TITLE
142 implement non strict equality compar

### LINKED ISSUE #142: Implement non strict equality comparison

### PR DESCRIPTION
**Pull request by JTeeuwissen**
resolved #142

### CODE REVIEW

File: core/test/Operation/Boolean/TestEquality.hs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import JsonLogic.Pure.Evaluator
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase)
+import qualified Test.Tasty.HUnit as U
+
+-- See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality
+equalityUnitTests :: TestTree
+equalityUnitTests =
+  testGroup
+    "Read json unit tests"
```
Comment (musicismyalibi) [inline core/test/Operation/Boolean/TestEquality.hs:15]: I think this has not yet been changed to the correct thing after copying.

**Review by musicismyalibi — COMMENTED** [review]
I think the tests for the inequality operation are missing.

File: core/test/Operation/Boolean/TestEquality.hs (hunk 1 above)
Comment (JTeeuwissen) [reply core/test/Operation/Boolean/TestEquality.hs:15]:
```suggestion
    "Equality unit tests"
```

File: core/src/JsonLogic/Operation/Boolean.hs [hunk 2]
```diff
@@ -36,14 +36,57 @@ evaluateIf _ _ _ = throwError "Wrong number of arguments for if"
 -- Helper functions
 
 evaluateLogic :: Monad m => (Bool -> Bool -> Bool) -> Function m Json
-evaluateLogic operator evaluator (JsonArray [x, y]) vars = do
-  x' <- evaluateBool evaluator x vars
-  y' <- evaluateBool evaluator y vars
-  return $ JsonBool $ x' `operator` y'
+evaluateLogic operator evaluator (JsonArray [x', y']) vars = do
```
Comment (GerardvSchie) [inline core/src/JsonLogic/Operation/Boolean.hs:39]: Better to have x and y as initial. and then x' and y' as evaluated values.

File: core/src/JsonLogic/Operation/Boolean.hs [hunk 3]
```diff
@@ -36,14 +36,57 @@ evaluateIf _ _ _ = throwError "Wrong number of arguments for if"
...
+-- | Evaluate loose equals
+looseEquals :: Monad m => Function m Json
+looseEquals evaluator (JsonArray [x', y']) vars = do
+  x <- evaluator x' vars
+  y <- evaluator y' vars
+  return $ JsonBool $ looseEq x y
+looseEquals _ _ _ = throwError "Wrong number of arguments for loose not equals operator"
+
+looseNotEquals :: Monad m => Function m Json
+looseNotEquals evaluator (JsonArray [x', y']) vars = do
```
Comment (GerardvSchie) [inline core/src/JsonLogic/Operation/Boolean.hs:60]: here too

**Review by GerardvSchie — CHANGES_REQUESTED** [review]

**Pull request merged**